salt/modules/ps.py

```diff
@@ -64,7 +64,18 @@ def _get_proc_name(proc):
 
     It's backward compatible with < 2.0 versions of psutil.
     '''
-    return proc.name() if PSUTIL2 else proc.name
+    ret = []
+    if PSUTIL2:
+        try:
+            ret = proc.name()
+        except (psutil.NoSuchProcess, psutil.AccessDenied):
+            pass
+    else:
+        try:
+            ret = proc.name
+        except (psutil.NoSuchProcess, psutil.AccessDenied):
+            pass
+    return ret
```

Comment:
Way better! :)

No idea why I didn't do it like this in the first place. Updated the PR accordingly.